src/iter/collect/mod.rs

```diff
@@ -129,14 +130,16 @@ impl<'c, T: Send + 'c> Collect<'c, T> {
     /// and return a slice to the uninitialized tail of the vector
     ///
     /// Safety: The tail slice is uninitialized
-    unsafe fn reserve_get_tail_slice(vec: &mut Vec<T>, len: usize) -> &mut [T] {
+    unsafe fn reserve_get_tail_slice(vec: &mut Vec<T>, len: usize) -> &mut [MaybeUninit<T>] {
```

Comment:
I don't think we need to mark this function as `unsafe` anymore. This has the added benefit of shrinking the `unsafe` block in `Collect::with_consumer` down to setting the final vector length when the collect completes.

Reply:
Makes sense, done.